parts/k8s/cloud-init/artifacts/etcd.service

```diff
@@ -2,9 +2,8 @@
 Description=etcd - highly-available key value store
 Documentation=https://github.com/coreos/etcd
 Documentation=man:etcd
-After=network.target
-Wants=network-online.target
-RequiresMountsFor=/var/lib/etcddisk
+After=network.target var-lib-etcddisk.mount
+Wants=network-online.target var-lib-etcddisk.mount
```

Comment:
I found the discussion of systemd dependency fields interesting this morning, thanks @Michael-Sinz and @jackfrancis. So I understand this change, except for why `var-lib-etcddisk.mount` needs to be in the `Wants=` stanza. Why isn't it being in the `After=` stanza sufficient?

Otherwise lgtm++.

Reply:
Ok, so the After is all that is needed (if you read my bug report: https://github.com/Azure/aks-engine/issues/3807 you can see that)

However, if someone were to manually start etcd from some unknown state, After does not make sure the mount exists, so the Wants= says add the mount to the "must start" (After just means "after it ran")

Note that Wants= is not after, it just means "add to the list of things to start" 

As I said in the bug report, the only things required to make it work for the boot scenarios is the After= and the removal of the RequiresMountsFor= - but that does not make it semantically the same for manual/other systemd operations.

It does not cost much to add the other two items (steps 3 and 4 from the bug report) and it makes this basically semantically the same as before but mechanically different (and thus fixing the problem)

Reply:
> Should be optional but I can imagine edge cases where it could be needed

As you said in the bug report, yep. Thanks for re-explaining!